src/rubocop/OffenseFormatter.js

```diff
@@ -0,0 +1,57 @@
+'use babel'
+
+import { satisfies } from 'semver'
+import getRuleDocumentation from './helpers/documentation-cache'
+
+import ErrorFormatter from '../ErrorFormatter'
+
+const SEVERITY_MAPPING = {
+  refactor: 'info',
+  convention: 'info',
+  warning: 'warning',
+  error: 'error',
+  fatal: 'error',
+}
+
+const HASCOPNAME_VERSION_RANGE = '>=0.52.0 <0.68.0'
+const RULE_MATCH_REGEX = /https:\/\/.*#(.*)/g
+
+export default class OffenseFormatter extends ErrorFormatter {
+  format(version, {
+    message: rawMessage, location, severity, cop_name: copName,
+  }, filePath) {
+    const hasCopName = satisfies(version, HASCOPNAME_VERSION_RANGE)
+    const [excerpt, url] = rawMessage.split(/ \((.*)\)/, 2)
+    let position
+    if (location) {
+      const { line, column, length } = location
+      position = [[line - 1, column - 1], [line - 1, (length + column) - 1]]
+    } else {
+      position = this.topFileRange
+    }
+
+    const linterMessage = {
+      url,
+      excerpt: hasCopName ? excerpt : `${copName}: ${excerpt}`,
+      severity: SEVERITY_MAPPING[severity] || SEVERITY_MAPPING.error,
+      location: {
+        file: filePath,
+        position,
+      },
+    }
+
+    if (url) {
+      const ruleMatch = RULE_MATCH_REGEX.exec(url)
+      if (ruleMatch) {
+        const rule = ruleMatch[1]
+        getRuleDocumentation(rule).then((documentation) => {
```

Comment:
🚨 This will set the `description` property on the message only _after_ it has resolved from the cache. I'd be amazed if this worked for things that weren't hot in the cache. 

The `description` property of a message is already meant to [handle a `Promise` directly](https://github.com/steelbrain/linter/blob/833309a502d2ccfc2a1a6c20423cc8e8023205b3/docs/types/linter-message-v2.md). This allows you to pass in a function that only retrieves the documentation _if_ the user requests it. Currently you would be pulling in the documentation for every single rule, regardless of whether the user ever views it.

Reply:
I understand, but using `description: (() => getRuleDocumentation(rule) || '')`  shows the arrow to expand and see the documentation even when there is no documentation, I don't like it too much.